lib/usd/translators/jointWriter.cpp

```diff
@@ -254,25 +252,23 @@ static bool _FindDagPoseMembers(
 
 bool _GetLocalTransformForDagPoseMember(
     const MFnDependencyNode& dagPoseDep,
-    unsigned int             index,
+    unsigned int             logicalIndex,
     GfMatrix4d*              xform)
 {
     MStatus status;
 
     MPlug xformMatrixPlug = dagPoseDep.findPlug("xformMatrix");
-    if (index < xformMatrixPlug.numElements()) {
-        MPlug xformPlug = xformMatrixPlug[index];
+    MPlug xformPlug = xformMatrixPlug.elementByLogicalIndex(logicalIndex, &status);
```

Comment:
Right now I know there's only one call site and the only indices stuffed in here should the logical ones, but _just in case_, might be worth having debug assert to check if the logical index actually exists already via `getExistingArrayAttributeIndices` or similar.

Reply:
My understanding is that if somehow that logical index doesn't exist yet, it will be created... which, as a fallback, seems ok to me.
Though I guess it might make sense to just put in a debug-only check / warning message, since it WOULD be an odd/unexpected occurence, which might warrent further investigation...

Reply:
Ok, added a check...